### PR TITLE
Analyze TUI

### PR DESCRIPTION
**Pull request by mcastorina**
### Description:
TUI interface for the analyze sub-command.

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/analyzer/tui/form.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		inputs = append(inputs, textinputs.InputConfig{
+			Label:    "Shopify URL",
+			Key:      "url",
+			Required: true,
+		})
+	}
+
+	form := textinputs.New(inputs).
+		SetHeader(titleStyle.Render(fmt.Sprintf("Configuring %s analyzer", keyType))).
+		SetFooter("⚠️  Running TruffleHog Analyze will send a lot of requests ⚠️").
```
Comment (dustin-decker) [inline pkg/analyzer/tui/form.go:38]: Can we make this dependent on an attribute on the analyzers? Not all do this.

**Review by dustin-decker — APPROVED** [review]
LGTM barring a couple of non-blocking points. 
Can we support filtering by typing?

File: pkg/analyzer/tui/form.go (hunk 1 above)
Comment (mcastorina) [reply pkg/analyzer/tui/form.go:44]: Is there an existing attribute I can reference? If not, do you have a list of analyzers that do brute-force?

Also open to phrasing suggestions. We could change it to something like "Running TruffleHog Analyze may send a lot of requests"

File: pkg/analyzer/cli.go [hunk 2]
```diff
@@ -24,227 +27,75 @@ import (
...
-	case sourcegraphScan.FullCommand():
-		cfg.LogFile = analyzers.CreateLogFileName("sourcegraph")
-		sourcegraph.AnalyzeAndPrintPermissions(cfg, *sourcegraphKey)
-	case shopifyScan.FullCommand():
-		cfg.LogFile = analyzers.CreateLogFileName("shopify")
-		shopify.AnalyzeAndPrintPermissions(cfg, *shopifyKey, *shopifyStoreURL)
-	case opsgenieScan.FullCommand():
-		cfg.LogFile = analyzers.CreateLogFileName("opsgenie")
-		opsgenie.AnalyzeAndPrintPermissions(cfg, *opsgenieKey)
+	switch keyType {
```
[the remark this answers is not in the document]
Comment (mcastorina) [reply pkg/analyzer/cli.go:59]: When the TUI exits it returns the keytype and a secretinfo struct in this function, so this is the part that's responsible for actually running the analysis.

**Comment by mcastorina** [comment]
> LGTM barring a couple of non-blocking points. Can we support filtering by typing?

Filtering the list is supported but you have to hit `/` first. There's a help menu at the bottom of the screen (in gray, may be hard to spot).

You can also start the TUI via `trufflehog analyze keytype`.

**Comment by dustin-decker** [comment]
> > LGTM barring a couple of non-blocking points. Can we support filtering by typing?
> 
> Filtering the list is supported but you have to hit `/` first. There's a help menu at the bottom of the screen (in gray, may be hard to spot).
> 
> You can also start the TUI via `trufflehog analyze keytype`.

Ah okay. I misread it as being part of the previous one, like `j = down / filter`. Is the formatting there customizable? Maybe a tab would make it a bit more visually distinct

File: pkg/analyzer/tui/form.go (hunk 1 above)
Comment (dustin-decker) [reply pkg/analyzer/tui/form.go:44]: We'd have to look at the analyzers and define that attribute.

**Pull request merged**